src/libraries/System.Runtime/tests/System/ArgumentExceptionTests.cs

```diff
@@ -60,5 +60,32 @@ public static void Ctor_String_String_Exception()
             Assert.Contains(message, exception.Message);
             Assert.Contains(argumentName, exception.Message);
         }
+
+        [Fact]
+        public static void ThrowIfNullOrEmpty_ThrowsForInvalidInput()
+        {
+            AssertExtensions.Throws<ArgumentNullException>(null, () => ArgumentException.ThrowIfNullOrEmpty(null, null));
+            AssertExtensions.Throws<ArgumentNullException>("something", () => ArgumentException.ThrowIfNullOrEmpty(null, "something"));
+
+            AssertExtensions.Throws<ArgumentException>(null, () => ArgumentException.ThrowIfNullOrEmpty("", null));
+            AssertExtensions.Throws<ArgumentException>("something", () => ArgumentException.ThrowIfNullOrEmpty("", "something"));
+
+            ArgumentException.ThrowIfNullOrEmpty(" ");
+            ArgumentException.ThrowIfNullOrEmpty(" ", "something");
+            ArgumentException.ThrowIfNullOrEmpty("abc", "something");
+        }
+
+        [Fact]
+        public static unsafe void ThrowIfNullOrEmpty_UsesArgumentExpression_ParameterNameMatches()
```

Comment:
nit: `unsafe` seems to be redundant

```suggestion
        public static void ThrowIfNullOrEmpty_UsesArgumentExpression_ParameterNameMatches()
```